src/utils/cache/getWallet.ts

```diff
@@ -55,14 +49,16 @@ export const getWallet = async <TWallet extends EVMWallet>({
       wallet = await getLocalWallet({ chainId, walletAddress });
       break;
     default:
-      throw new Error(
+      throw createCustomError(
         `Wallet with address ${walletAddress} was configured with unknown wallet type ${walletDetails.type}`,
+        400,
+        "BAD_REQUEST",
```

Comment:
This is a low level helper func which really shouldn't know or imply anything about the caller's intent. So it shouldn't know what status code to set.